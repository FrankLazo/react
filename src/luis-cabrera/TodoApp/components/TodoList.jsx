import Todo from "./Todo";

const TodoList = ({ todos, setTodoEdit, todoDelete, todoToggleCompleted }) => {
	return (
		<section className="column is-8">
			<h2>Todo List</h2>

			{
				todos.length === 0
				? (
					<div>
						<p>No hay tareas, por favor agregar una!</p>
					</div>
				)
				: (
					todos.map(todo => (
						<Todo
							key={todo.id}
							todo={todo}
							todoDelete={todoDelete}
							todoToggleCompleted={todoToggleCompleted}
							setTodoEdit={setTodoEdit}
							/>
					))
				)
			}
		</section>
	)
}

export default TodoList;
