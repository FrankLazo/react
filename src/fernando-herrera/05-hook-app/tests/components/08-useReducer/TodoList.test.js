import { shallow } from 'enzyme';
import TodoList from '../../../components/08-useReducer/TodoList';
import demoTodos from '../../fixtures/demoTodos';

describe('<TodoList /> tests', () => {
	const handleDelete = jest.fn();
	const handleToggle = jest.fn();

	const wrapper = shallow(
		<TodoList
			todos={ demoTodos }
			handleToggle={ handleToggle }
			handleDelete={ handleDelete }
			/>
	);

	test('should show the right component', () => {
		expect(wrapper).toMatchSnapshot();
	});

	test('should have two elements with right arguments', () => {
		expect(wrapper.find('TodoListItem').length).toBe(demoTodos.length);
		expect(wrapper.find('TodoListItem').at(0).prop('handleToggle')).toEqual( expect.any(Function) );
		expect(wrapper.find('TodoListItem').at(0).prop('handleDelete')).toEqual( expect.any(Function) );
	});
});
