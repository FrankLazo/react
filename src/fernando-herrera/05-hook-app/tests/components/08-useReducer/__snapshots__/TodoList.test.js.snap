// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoList /> tests should show the right component 1`] = `
<ul>
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    index={0}
    key="1"
    todo={
      Object {
        "description": "Learn React",
        "done": true,
        "id": 1,
      }
    }
  />
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    index={1}
    key="2"
    todo={
      Object {
        "description": "Learn Mongo",
        "done": false,
        "id": 2,
      }
    }
  />
</ul>
`;
